core/src/main/java/org/hibernate/ogm/loader/nativeloader/impl/BackendCustomQuery.java

```diff
@@ -41,7 +41,8 @@
 	private final Set<String> querySpaces;
 	private final List<Return> customQueryReturns;
 
-	private final EntityKeyMetadata singleEntityKeyMetadata;
+	private EntityKeyMetadata singleEntityKeyMetadata;
```

Comment:
Can we make that final again? I'm rather keen on making things truely immutable which grants safe publication across threads.

You could create a container type to be returned by the `determine...` method if it is about the need for multiple return values (which I suppose).